### PR TITLE
Fix/elastic UI upgrade breaking changes in 7.11 and 7.12 3318

### LINKED ISSUE #3318: Elastic UI Upgrade Breaking Changes in 7.11 & 7.12
| Wazuh | Elastic | Rev |
| ----- | ------- | --- |
| 4.2   | 7.12     | 444 |

**Description**
EUI breaking changes errors in `Kibana 7.11 & 7.12`

**Steps to reproduce**

1. Go to Security Events
2. Open Browser Dev Tools
3. See error

**Screenshots**
image

**Additional context**
Additional information concerning these breaking changes can be found [here](https://elastic.github.io/eui/#/package/changelog)

### PR DESCRIPTION
**Pull request by asteriscos**
Hi team!

this fixes breaking changes in Elastic UI 7.12 upgrade:

- Updated legend and idSelected props of EuiButtonGroup to be required
- Removed EuiPopover's withTitle prop

Closes #3318 

To test:

- Go to Security Events
- Open Browser Dev Tools
- Check there's no EuiButtonGroup's error 

- Go to GDPR -> Controls
- Open Browser Dev Tools
- Check there's no EuiPopover's error

### CODE REVIEW

**Review by frankeros — APPROVED** [review]
CR: LGTM!

**Comment by frankeros** [comment]
please add changelog

File: public/controllers/agent/components/register-agent.js [hunk 1]
```diff
@@ -568,6 +571,7 @@ export class RegisterAgent extends Component {
               children: (
                 <EuiButtonGroup
                   color="primary"
+                  legend="Choose the architecture"
```
Comment (MauGaP) [inline public/controllers/agent/components/register-agent.js:574]: whats the difference between this "Chose the architecture" and the previous one on line 557?

Comment (asteriscos) [reply public/controllers/agent/components/register-agent.js:574]: Both legends represent the component `title`. It's for accessibility purposes. As you can see, both titles have the same text and so do their legends.

File: public/components/wz-search-bar/components/wz-search-badges/context-menu.tsx [hunk 2]
```diff
@@ -205,7 +205,7 @@ function EditFilterOperator(operator, setOperator) {
 
 function EditFilterConjuntion(conjuntion: string, setConjuntion): React.ReactNode {
   return <EuiFormRow label="Conjuntion">
-    <EuiButtonGroup options={[
+    <EuiButtonGroup legend="Conjunction" options={[
       { id: `conjuntion-AND`, label: "AND" },
       { id: `conjuntion-OR`, label: "OR" },
```
Comment (matiasmoreno876) [inline public/components/wz-search-bar/components/wz-search-badges/context-menu.tsx:210]: Could we pass these options to a variable?

Comment (asteriscos) [reply public/components/wz-search-bar/components/wz-search-badges/context-menu.tsx:210]: sounds good, let's improve the code readability

**Review by gabiwassan — APPROVED** [review]
LGTM !

**Review by matiasmoreno876 — APPROVED** [review]
LGTM

**Review by mpRegalado — APPROVED** [review]
CR: LGTM! :heavy_check_mark: 
Testing: LGTM! :heavy_check_mark:

**Review by sortizowlh — APPROVED** [review]
CR:  ✔️
Testing: 7.11.2 ✔️

**Pull request merged**